### PR TITLE
Use non-root for python-base

### PR DESCRIPTION
**Pull request by david-caro**
* This allows using the venv from outside the containers.
* Avoids ugly root owned files on the code directory.

Signed-off-by: David Caro <david@dcaro.es>

### CODE REVIEW

**Comment by david-caro** [comment]
@jacquerie this works on linux, but on macos it should be just as if it was before, but instead of the files generated owned by root, they will be owned by user '1000', can you try it out?

I'll add a bit of magic to change ownership of the virtualenv at runtime, so we don't have to change it manually for the first time if it already exists.

**Comment by david-caro** [comment]
Okok, this new version allows you to define (with the pr inspirehep/inspire-next#2124) to use non-root even on mac, restoring back the permissions of the venv to whatever the env variables `BASE_USER_UID` and `BASE_USE R_GID` have, using only `BASE_USER_UID` if the other is not defined. It also fixes the rights of the venv on entering the container, so no need to manually change anything even if you have already a venv from previous runs on your system.

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,12 @@ services:
...
+    - DOCKER_DATA=/tmp/data
+    - DOCKER_VERSION=1.12.3-0~trusty
+    - DOCKER_COMPOSE_VERSION=1.9.0
+    - secure: B2o+OVhyzRFr0SvLl18FtTT6DO6+d69BizgULki5MTgMdxGJrVAEoj+AAl5edwJwjN6MZ4OiSCLoauKHjjlHjZT2A0J8wMHGCAD3IpdpZ9wacw0qjkNKkUXDRTdFycDXVTngHn0YJ++tA7kHG98DAkRq7YfWrsb5kxxFo54Gl8UkvWpOxKx0nd6n1d4/OooKYhSXaPKPcZ/Mpa/LR1/O2xC61RBYrzk3DPlFmW5+4F2iSKh7ALPd92joyDsxRnOuVcQU/aCb6fUPwVH7znn2Rw9i8pwCYNXoJdNzM0DvyV4tFf1aigsHS7uGRFtzXGSe57MEoJl7prCbHKjHLZ+ZrOgay4CmYmpdEAYKTJwNPdM1mm80wftYfyo3kMynDoJU82wvGwlQ4ZDYckvJ11XVKASLO40QNxLiZaOCs++R4WExVbK2yNGMUyoC26Z7pJl2t5FGEdBjr73Ee19NslwFgGBsI5IE1JG7SToFFVaQiwIKuuTEQX81JZFsihpv8SgaAFj2Ur6D50yh4YnUbau7f2wei6xHlKg+GGYQsMweYxXQUrAzdr5igOrb3AJ3dW8h9vH6LIERlblImbvAFu0xFl0dK1Bbv6TkB908DVtN1JzrcN1iu9ca6cXyIfuSkR8cBQ/DK3Ih8s/V1slq1Qf71ydt1a+TVT95rtq1StnneD0=
```
Comment (david-caro) [inline .travis.yml]: I'm not sure where this comes from :/

File: .travis.yml [hunk 2]
```diff
@@ -38,6 +39,7 @@ matrix:
 
 script:
   - chmod +x *.sh
+  - mkdir -p "$DOCKER_DATA"
```
Comment (david-caro) [inline .travis.yml]: This is not really needed

File: build.sh [hunk 3]
```diff
@@ -70,6 +70,15 @@ parse_options(){
 }
 
 
+build_venv_rights_fixer() {
+    cd python_base
+    gcc -o fix_venv_rights fix_venv_rights.c
+    local res=$?
+    cd -
```
Comment (david-caro) [inline build.sh]: Redirect the output to null, no need for the extra clutter

File: test.sh [hunk 4]
```diff
@@ -76,6 +76,11 @@ install_docker_compose(){
 clone_next(){
     rm -rf "$NEXT_DIR"
     git clone https://github.com/inspirehep/inspire-next.git "$NEXT_DIR"
+    # TODO remove once it's merged
+    cd inspire-next
+    git fetch origin refs/pull/2124/head
+    git reset --hard FETCH_HEAD
+    cd -
```
Comment (david-caro) [inline test.sh]: It's actually not really needed, as it can go on without the extra vars, so remove.

File: python_base/fix_venv_rights.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#include <stdlib.h>
```
Comment (david-caro) [inline python_base/fix_venv_rights.c]: Explain on a comment on top why is needed.

File: python_base/docker_entrypoint.sh [hunk 6]
```diff
@@ -21,19 +21,76 @@
...
+    if ! [[ -f "$VENV_PATH/bin/activate" ]]; then
+        prepare_venv
+    else
+        source "$VENV_PATH"/bin/activate
+    fi
+
+    find \( -name __pycache__ -o -name '*.pyc' \) -delete
+
+    trap forward_sigterm SIGTERM
+    trap forward_sigterm SIGINT
```
Comment (david-caro) [inline python_base/docker_entrypoint.sh]: s/sigterm/sigint/

File: python_base/docker_entrypoint.sh [hunk 7]
```diff
@@ -21,19 +21,76 @@
...
+
+    "$@" &
+    child="$!"
+    fg 2 >/dev/null
```
Comment (david-caro) [inline python_base/docker_entrypoint.sh]: Redirecting to null seems to have not effect here.

**Comment by david-caro** [comment]
Big issue though, running `docker-compose up` ends up with the python_base containers complaining with:
```
worker_1        | bash: cannot set terminal process group (-1): Inappropriate ioctl for device
worker_1        | bash: no job control in this shell
worker_1        | "$@"
worker_1        | bash: celery: command not found
worker_1        | Restoring permissions or venv to 1000:1000
inspirenext_worker_1 exited with code 127
```

File: python_base/docker_entrypoint.sh [hunk 8]
```diff
@@ -21,19 +21,76 @@
...
+# monitor mode (-m)
+set -me
+
+VENV_PATH=/virtualenv
+
+
+restore_venv_rights() {
+    if [[ "$BASE_USER_UID" != "" ]]; then
+        BASE_USER_GID="${BASE_USER_GID:-$BASE_USER_UID}"
+        echo "Restoring permissions or venv to $BASE_USER_UID:$BASE_USER_GID"
```
Comment (david-caro) [inline python_base/docker_entrypoint.sh]: s/or/of/

File: python_base/docker_entrypoint.sh (hunk 7 above)
Comment (david-caro) [inline python_base/docker_entrypoint.sh]: Hmmm.... if using fg, not sure if the forwarding of the signals is needed (with just wait it's needed for sure), have to investigate.

Comment (david-caro) [inline python_base/docker_entrypoint.sh]: That would simplify the script :)

File: python_base/fix_venv_rights.c (hunk 5 above)
Comment (jacquerie) [inline python_base/fix_venv_rights.c]: Yes, I'm curious to see why you need this O_O

File: .travis.yml (hunk 1 above)
Comment (david-caro) [inline .travis.yml]: This is just indenting it properly.

**Pull request merged**